cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/internal/replication/ReplicationTestBase.java

```diff
@@ -238,14 +240,30 @@ protected void assertCookieInterceptorPresent(ReplicatorBuilder p, String expect
         Assert.assertEquals(CookieInterceptor.class, reqIList.get(1).getClass());
         Assert.assertEquals(1, respIList.size());
         Assert.assertEquals(CookieInterceptor.class, respIList.get(0).getClass());
-        CookieInterceptor ci = (CookieInterceptor)reqIList.get(1);
-        Field srbField = CookieInterceptor.class.getDeclaredField("sessionRequestBody");
+        CookieInterceptorBase ci = (CookieInterceptorBase)reqIList.get(1);
+        Field srbField = CookieInterceptorBase.class.getDeclaredField("sessionRequestBody");
         srbField.setAccessible(true);
         byte[] srb = (byte[])srbField.get(ci);
         String srbString = new String(srb);
         Assert.assertEquals(expectedRequestBody, srbString);
     }
 
+    protected void assertIamCookieInterceptorPresent(ReplicatorBuilder p)
+            throws NoSuchFieldException, IllegalAccessException {
+        // peek inside these private fields to see that interceptors have been set
+        Field reqI = ReplicatorBuilder.class.getDeclaredField("requestInterceptors");
+        Field respI = ReplicatorBuilder.class.getDeclaredField("responseInterceptors");
+        reqI.setAccessible(true);
+        respI.setAccessible(true);
+        List<HttpConnectionRequestInterceptor> reqIList = (List)reqI.get(p);
+        List<HttpConnectionRequestInterceptor> respIList = (List)respI.get(p);
+        // This relies on the UserAgentInterceptor being added before the CookieInterceptor.
```

Comment:
Comment needs a slight update from the copy/paste to `IamCookieInterceptor`.

Reply:
Done in 521ecd56